appinventor/blocklyeditor/src/component_database.js

```diff
@@ -513,28 +515,43 @@ Blockly.ComponentDatabase.EVENTDESC = /EventDescriptions$/;
 Blockly.ComponentDatabase.prototype.populateTranslations = function(translations) {
   var newkey;
   for (var key in translations) {
-    if (translations.hasOwnProperty(key)) {
-      var parts = key.split('-', 2);
-      if (parts[0] === 'COMPONENT') {
-        this.i18nComponentTypes_[parts[1]] = translations[key];
-      } else if (parts[0] === 'PROPERTY') {
-        this.i18nPropertyNames_[parts[1]] = translations[key];
-      } else if (parts[0] === 'EVENT') {
-        this.i18nEventNames_[parts[1]] = translations[key];
-      } else if (parts[0] === 'METHOD') {
-        this.i18nMethodNames_[parts[1]] = translations[key];
-      } else if (parts[0] === 'PARAM') {
-        this.i18nParamNames_[parts[1]] = translations[key];
-      } else if (parts[0] === 'EVENTDESC') {
-        newkey = parts[1].replace(Blockly.ComponentDatabase.EVENTDESC, '');
-        this.i18nEventDescriptions_[parts[1]] = translations[key];
-      } else if (parts[0] === 'METHODDESC') {
-        newkey = parts[1].replace(Blockly.ComponentDatabase.METHODDESC, '');
-        this.i18nMethodDescriptions_[newkey] = translations[key];
-      } else if (parts[0] === 'PROPDESC') {
-        newkey = parts[1].replace(Blockly.ComponentDatabase.PROPDESC, '');
-        this.i18nPropertyDescriptions_[newkey] = translations[key];
-      }
+    if (!translations.hasOwnProperty(key)) {
+      continue;
+    }
+    var parts = key.split('-', 2);
+    var type = parts[0];
+    var jsKey = parts[1];
+    var translation = translations[key];
+    switch(type) {
+      case 'COMPONENT':
+        this.i18nComponentTypes_[jsKey] = translation;
+        break;
+      case 'PROPERTY':
+        this.i18nPropertyNames_[jsKey] = translation;
+        break;
+      case 'EVENT':
+        this.i18nEventNames_[jsKey] = translation;
+        break;
+      case 'METHOD':
+        this.i18nMethodNames_[jsKey] = translation;
+        break;
+      case 'PARAM':
+        this.i18nParamNames_[jsKey] = translation;
+        break;
+      case 'EVENTDESC':
+        this.i18nEventDescriptions_[jsKey] = translation;
+        break;
+      case 'METHODDESC':
+        this.i18nMethodDescriptions_[jsKey] = translation;
+        break;
+      case 'PROPDESC':
+        this.i18nPropertyDescriptions_[jsKey] = translation;
+        break;
+      case 'OPTION':
+        this.i18nOptionNames_[jsKey] = translation;
+        break;
+      case 'OPTIONLIST':
+        this.i18nOptionListTags_[jsKey] = translation;
```

Comment:
Slightly pedantic but I would recommend including the `break;` here even though it is redundant.